src/main/kotlin/com/pubnub/api/PNConfiguration.kt

```diff
@@ -251,6 +251,20 @@ open class PNConfiguration(
      */
     var sendStateWithHeartbeat = true
 
+    /**
+     * When `true` the SDK will resend the last channel state that was set using [PubNub.setPresenceState]
+     * for the current [userId] when reconnecting subscriptions (i.e. subscribe calls with timetoken == 0).
+     *
+     * Applies only if [enableEventEngine] is true.
+     *
+     * Defaults to `true`.
+     *
+     * Please note that `sendStateWithSubscribe` doesn't apply to state that was set on channel groups.
+     * It is recommended to disable this option if you set state for channel groups using [PubNub.setPresenceState],
+     * otherwise that state may be overwritten on the next subscribe reconnection with individual channel states.
+     */
+    var sendStateWithSubscribe = true
+
     /**
```

Comment:
This is side comment.

As we discussed. If possible I would be beneficial to add tests that demonstrate purpose or sending state with subscribe tt=0 and hearbeat.